bike.js

```diff
@@ -0,0 +1,145 @@
+const Table = require('cli-table2')
+
+
+async function _getUserBikedToWorkDistance(state, userId) {
+  const rows = await state.db.query('SELECT * FROM users WHERE id = $1', [userId + ''])
+  if (rows.length !== 1) return null
+  const user = rows[0]
+  return user.biked_to_work_distance
+}
+
+async function _setUserBikedToWorkDistance(state, message, distanceMeters) {
+  const userId = message.author.id
+  await state.db.query('UPDATE users SET biked_to_work_distance = $1 WHERE id = $2', [distanceMeters, userId + ''])
+  message.raw.tags.push('#bike-distance-saved')
+}
+
+async function _hasUserBikedToWorkBefore(state, userId) {
+  const rows = await state.db.query('SELECT COUNT(*) FROM bike_runs WHERE user_id = $1', [userId + ''])
+  return parseInt(rows[0].count, 10) > 0
+}
+
+async function _handleDistanceInputAbsentForBikeDone(message, state, hasBikedToWorkBefore) {
```

Comment:
Why do you call this function `Absent` and the function below `Present`? Could you explain the logic behind the choice of names please :)

Reply:
The "I biked to work" signal from the chat app can take some different formats, like:
- `#btw`
- `#btw 5`
- `#btw 2 #save`

where 5 and 2 represent the number of kilometers biked, but the value is not required. There are cases to handle when this distance input is present and cases to handle when it is absent, hence, the naming.